app/services/publisher_statement_getter.rb

```diff
@@ -1,53 +1,100 @@
-          transaction["created_at"].to_date.at_beginning_of_month == cutoff
-        }
-      when "last_month"
-        cutoff = (Date.today - 1.month).at_beginning_of_month
-        transactions.select { |transaction|
-          transaction["created_at"].to_date.at_beginning_of_month == cutoff
-        }
-      else
-        transactions
-    end
-  end
-
   def replace_account_identifiers_with_titles(transactions)
-    transactions.map { |transaction|
+    transactions.map do |transaction|
       account_identifier = transaction["channel"]
       if account_identifier.starts_with?(Publisher::OWNER_PREFIX)
-        transaction["channel"] = "All"
+        transaction["channel"] = I18n.t("publishers.statements.index.account")
       elsif account_identifier.blank?
         transaction["channel"] = "Manual"
       else
-        channel = Channel.find_by_channel_identifier(account_identifier)
-        transaction["channel"] = channel&.publication_title || account_identifier
+        transaction["channel"] = channel_name(account_identifier)
       end
-      transaction
-    }
+
+      Statement.new(
+        channel: transaction["channel"],
+        description: transaction["description"],
+        transaction_type: transaction["transaction_type"],
+        amount: transaction["amount"]&.to_d,
+        settlement_currency: transaction["settlement_currency"],
+        settlement_amount: transaction["settlement_amount"]&.to_d,
+        settlement_destination_type: transaction["settlement_destination_type"],
+        settlement_destination: transaction["settlement_destination"],
+        created_at: transaction["created_at"].to_date,
+      )
+    end
+  end
+
+  def get_uphold_transactions
+    uphold = []
+
+    publisher.uphold_connection.uphold_connection_for_channels.each do |card_connection|
+      transactions = card_connection.uphold_connection.uphold_client.transaction.all(id: card_connection.card_id)
+      next if transactions.blank?
+
+      transactions.each do |transaction|
+        uphold << Statement.new(
+          channel: card_connection.channel.details.publication_title,
+          transaction_type: Statement::UPHOLD_CONTRIBUTION,
+          amount: transaction.origin.dig("amount")&.to_d,
+          settlement_currency: transaction.destination.dig("currency"),
+          settlement_amount: transaction.destination.dig("amount")&.to_d,
+          created_at: transaction.createdAt.to_date,
+        )
+      end
+    end
+
+    # Replicate the similar behavior of Eyeshade. Aggregates all the contribution amounts for the month, and puts them into one entry
+    uphold.group_by { |u| u.created_at.at_end_of_month }.each do |date, entries|
+      entries.group_by { |e| e.channel }.each do |channel, channel_entries|
+        amount = channel_entries.sum { |x| x.amount }
+
+        uphold << Statement.new(
+          channel: channel,
+          transaction_type: Statement::UPHOLD_CONTRIBUTION_SETTLEMENT,
+          amount: -amount,
```

Comment:
Why is this negative? (Preferable you add a comment in-line to explain)

Reply:
https://github.com/brave-intl/publishers/pull/2209/commits/35058de3041adc5cf544b4d3bce3a4d4594a66da